source/MaterialXCore/Node.cpp

```diff
@@ -634,6 +647,33 @@ bool NodeGraph::validate(string* message) const
             validateRequire(getOutputCount() == nodeDef->getActiveOutputs().size(), res, message, "NodeGraph implementation has a different number of outputs than its NodeDef");
         }
     }
+    // Check interfaces on nodegraphs which are not definitions
+    if (!hasNodeDefString())
```

Comment:
Check interfacename input existence and if that input has connections that they exist as well.